nss-day-cohort-30/bangazon-site-mighty-mahi-mahi
# View product details, complete orders, reduce product quantity

Fixes:
- If user clicks on product details hyperlink, the product details are displayed.
- If the user clicks on add to cart while viewing the product details the product is added to the users cart
- If the user does not have an open order and they add a product to their cart, a new order is created and the products is added to their cart
- When the user is viewing their shopping cart they have the option of choosing, from a drop-down menu, one of their payment types in order to complete their open order. 
- If the user chooses a payment type and clicks complete order, the user's open order is updated to reflect date/time completed as well as payment type. The product quantity is reduced by the number of units purchased in the completed order. 

Changes proposed in this request & reasons behind organizational or architectural decisions:
-Changed OrderLineItem property "Cost" to be named "Total" -> represents total price customer will pay for chosen quantity of product. 

Steps to test:
- Start Bangazon
- Click on any product details
- Add product to cart
- Click shopping cart in navigation bar
- Select payment type
- Click complete purchase

System configuration (3rd party libraries to be installed, command line utilities to run, UI instructions, expected outcome):
-N/A

Commit message:
- changed Home nav bar link to show Products Index View
- product detail page has correct information.
- add to cart now redirects back to Products/Index/View
- users can add qty 1 of a product to their shopping cart from the product details view
- user can view shopping correct shopping cart if they have an open order, need to fix error if no order has been opened
- order can be completed by adding a payment time and hitting complete purchase button in shopping cart
- product qty is decreased by number of items purchased

Link to feature ticket:
-https://github.com/nss-day-cohort-30/bangazon-site-mighty-mahi-mahi/issues/1

@mighty-mahi-mahi
